apps/wallet/tests/sites-to-cs-messaging.spec.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type Page } from '@playwright/test';
+
+import { expect, test } from './fixtures';
+import { createWallet } from './utils/auth';
+
+function getInAppMessage(page: Page, id: string) {
+    return page.evaluate(
+        (anId) =>
+            Promise.race([
+                new Promise((r) => setTimeout(() => r(null), 1000)),
+                new Promise((r) => {
+                    const callBackFN = (msg: MessageEvent) => {
+                        if (
+                            msg.data.target === 'sui_in-page' &&
+                            msg.data.payload.id === anId
+                        ) {
+                            window.removeEventListener('message', callBackFN);
+                            r(msg.data.payload);
+                        }
+                    };
+                    window.addEventListener('message', callBackFN);
+                }),
+            ]),
+        id
+    );
+}
+
+const noPermissionError = {
+    payload: {
+        error: true,
+        message:
+            "Operation not allowed, dapp doesn't have the required permissions",
+    },
+};
+
+const noAccountError = {
+    payload: {
+        error: true,
+        message: "Cannot read properties of undefined (reading 'account')",
+    },
+};
+
+test.describe('site to content script messages', () => {
+    test.beforeAll(async ({ page, extensionUrl }) => {
+        await createWallet(page, extensionUrl);
+        await page.close();
+    });
+
+    const allTests = [
+        ['get accounts', { type: 'get-account' }, noPermissionError],
+        [
```

Comment:
I'd much rather a test that validated the wallet standard interface instead of our underlying message protocol (except for the security cases testing that we can't do keyring things).

Reply:
Tried to use the wallet interface but is difficult to get a reference to it. Tried to inject adapter code but no luck. Any suggestions?

Reply:
added a demo dapp and testing some scenarios through it but kept some of the tests here as well to make sure that bypassing wallet interface and trying to access user data is still blocked